python/lsst/ts/Dome/mock_controller.py

```diff
@@ -143,120 +211,238 @@ async def cmd_loop(self, reader, writer):
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain keys for all lower level
+            components to be configured with values that are dicts with keys for all the parameters that
+            need to be configured. The structure is
+            "AMCS":
+                "jmax"
+                "amax"
+                "vmax"
+            "LWSCS":
+                "jmax"
+                "amax"
+                "vmax"
+
+            It is assumed that the values of the configuration parameters are validated to lie within the
+            limits before being passed on to this function.
+        """
         self.log.info(f"Received command 'config' with arguments {kwargs}")
+        if kwargs["AMCS"]:
+            amcs_config = kwargs["AMCS"]
+            if amcs_config["jmax"]:
+                self.amcs.jmax = amcs_config["jmax"]
+            if amcs_config["amax"]:
+                self.amcs.amax = amcs_config["amax"]
+            if amcs_config["vmax"]:
+                self.amcs.vmax = amcs_config["vmax"]
+        if kwargs["LWSCS"]:
+            lwscs_config = kwargs["LWSCS"]
+            if lwscs_config["jmax"]:
+                self.lwscs.jmax = lwscs_config["jmax"]
+            if lwscs_config["amax"]:
+                self.lwscs.amax = lwscs_config["amax"]
+            if lwscs_config["vmax"]:
+                self.lwscs.vmax = lwscs_config["vmax"]
 
-    async def quit(self):
-        self.log.info("Received command 'quit'")
-        await self.stop()
+    async def park(self):
+        """Mock parking the dome.
+        """
+        self.log.info(f"Received command 'park'")
+        await self.amcs.park()
+
+    async def setTemperature(self, **kwargs):
+        """Mock setting the preferred temperature in the dome.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain the key "temperature" with a
+            float value.
+        """
+        self.log.info(f"Received command 'setTemperature' with arguments {kwargs}")
+        await self.thcs.setTemperature(temperature=float(kwargs["temperature"]))
+
+    async def fans(self, **kwargs):
+        """Mock switching on or off the fans in the dome.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain the key "action" with a
+            string value (ON or OFF).
+        """
+        self.log.info(f"Received command 'fans' with arguments {kwargs}")
+
+    async def inflate(self, **kwargs):
+        """Mock inflating or deflating the inflatable seal.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain the key "action" with a
+            string value (ON or OFF).
+        """
+        self.log.info(f"Received command 'inflate' with arguments {kwargs}")
```

Comment:
Please consider recording the commanded action somehow, e.g. `self.seal_inflated = action == "ON"` if you don't need a mock device.

Reply:
Good point and I actually didn't consider implementing a mock handler for this until now. I have added a couple of methods for this and corresponding unit tests.